package/MDAnalysis/coordinates/GRO.py

```diff
@@ -352,21 +378,26 @@ def write(self, selection, frame=None):
                         resname=resname,
                         index=truncated_atom_index,
                         name=name,
-                        pos=coordinates[atom_index]
+                        pos=positions[atom_index]
                     ))
 
+            try:  # for AtomGroup/Universe
+                tri_dims = obj.universe.coord.triclinic_dimensions
+            except AttributeError:  # for Timestep
+                tri_dims = obj.triclinic_dimensions
+
             # Footer: box dimensions
-            if np.all(u.trajectory.ts.dimensions[3:] == [90., 90., 90.]):
+            if np.all(selection.dimensions[3:] == [90., 90., 90.]):
```

Comment:
I think it's better to use `np.allclose(...)` here to avoid floating round of errors.